src/explorer/trialApp/TrialAppTreeItem.ts

```diff
@@ -0,0 +1,100 @@
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { RequestError } from 'request-promise/errors';
+import { AzExtTreeItem, IActionContext } from 'vscode-azureextensionui';
+import { localize } from '../../localize';
+import { openUrl } from '../../utils/openUrl';
+import { requestUtils } from '../../utils/requestUtils';
+import { AzureAccountTreeItem } from '../AzureAccountTreeItem';
+import { ISiteTreeItem } from '../ISiteTreeItem';
+import { SiteTreeItemBase } from '../SiteTreeItemBase';
+import { ITrialAppMetadata } from './ITrialAppMetadata';
+
+export class TrialAppTreeItem extends SiteTreeItemBase implements ISiteTreeItem {
+    public static contextValue: string = 'trialApp';
+    public contextValue: string = TrialAppTreeItem.contextValue;
+
+    public metadata: ITrialAppMetadata;
+
+    public defaultHostName: string;
+
+    public defaultHostUrl: string;
+
+    private readonly _defaultHostName: string;
+
+    public get label(): string {
+        return this.metadata.siteName ? this.metadata.siteName : localize('nodeJsTrialApp', 'NodeJS Trial App');
+    }
+
+    private get minutesLeft(): number {
+        return (this.metadata.timeLeft / 60);
+    }
+
+    public get description(): string {
+        return isNaN(this.minutesLeft) ?
+            localize('expired', 'Expired') : `${this.minutesLeft.toFixed(0)} ${localize('minutesRemaining', 'min. remaining')}`;
+    }
+
+    public get id(): string {
+        return `trialApp${this._defaultHostName}`;
+    }
+
+    private constructor(parent: AzureAccountTreeItem, metadata: ITrialAppMetadata) {
+        super(parent);
+        this.metadata = metadata;
+        this.defaultHostName = this.metadata.hostName;
+        this.defaultHostUrl = `https://${this.defaultHostName}`;
+    }
+
+    public static async createTrialAppTreeItem(parent: AzureAccountTreeItem, loginSession: string): Promise<TrialAppTreeItem> {
+        const metadata: ITrialAppMetadata = await this.getTrialAppMetaData(loginSession);
+        return new TrialAppTreeItem(parent, metadata);
+    }
+
+    public static async getTrialAppMetaData(loginSession: string): Promise<ITrialAppMetadata> {
+        const metadataRequest: requestUtils.Request = await requestUtils.getDefaultRequest('https://tryappservice.azure.com/api/vscoderesource', undefined, 'GET');
+
+        metadataRequest.headers = {
+            accept: "*/*",
+            "accept-language": "en-US,en;q=0.9",
+            "sec-fetch-dest": "empty",
+            "sec-fetch-mode": "cors",
+            "sec-fetch-site": "same-origin",
+            cookie: `loginsession=${loginSession}`
+        };
+
+        try {
+            const result: string = await requestUtils.sendRequest<string>(metadataRequest);
+            return <ITrialAppMetadata>JSON.parse(result);
+        } catch (e) {
```

Comment:
I'm not really sure what the value added here is.  Are the errors that Azure throws really human unfriendly?

Reply:
Yeah, I see what you're getting at. They are a little unfriendly, however in the event that an error occurs it would probably be more useful to just pass the error to the user.